### PR TITLE
chore(developer, web): developer host page linkage to modularized KMW 🧩

### PR DESCRIPTION
**Pull request by jahorton**
This PR aims to reconnect Web within Developer for its keyboard test-host page, maintaining 16.0 functionality.  There's also a bit of build-script patchup involved due to the shift in target names.

While we could set a user test here, we kind of want to let it merge early so that the CI build checks for Developer stop showing up red.  We can always set the user test with all the other user tests that'll be against the base branch.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by jahorton** [comment]
OK.  We have successful bundling, by the look of it... but...

```
Uncaught TypeError: keyman.isPositionSynthesized is not a function
    updateLogCursor http://user-pc:8008/chargrid.js:76
    logContent http://user-pc:8008/chargrid.js:54
    <anonymous> http://user-pc:8008/chargrid.js:122
    <anonymous> http://user-pc:8008/chargrid.js:124
```

```
TypeError: keyman.alignInputs is not a function
    onload test.js:245
    callEvent legacyEventEmitter.ts:142
    KeymanEngine3 keymanEngine.ts:135
    emit index.js:181
    ContextManagerBase contextManagerBase.ts:277
    step keymanweb.js:5226
    verb keymanweb.js:5173
    fulfilled keymanweb.js:5144
[bundle.min.js:2:14159](http://user-pc:8008/lib/sentry/bundle.min.js)
    mn bundle.min.js:2
    callEvent legacyEventEmitter.ts:144
    KeymanEngine3 keymanEngine.ts:135
    emit index.js:181
    ContextManagerBase contextManagerBase.ts:277
    step keymanweb.js:5226
    verb keymanweb.js:5173
    fulfilled keymanweb.js:5144
```

Looks like I didn't catch all the needed hooks the first time.

**Comment by jahorton** [comment]
Looks like everything is properly patched up now; I did a little personal testing with the host page on the build artifact version.

File: developer/src/server/build.sh [hunk 1]
```diff
@@ -124,20 +124,18 @@ fi
 
 if (( build_keymanweb )); then
   pushd "$KEYMAN_ROOT/web/"
-  ./build.sh --no-minify
+  ./build.sh build
```
Comment (mcdurdin) [inline developer/src/server/build.sh:127]: Keyman Developer used the unminified version of KeymanWeb in the past, to make it easier to debug keyboards in JS if needed (which helped for 'imx' style keyboards in particular). Would this still be helpful now? As in, should we add `--debug` here?

File: developer/src/server/src/site/chargrid.js [hunk 2]
```diff
@@ -73,17 +73,10 @@
...
-      selStart = 0;
-      selLength = 0;
-      selDirection = 'forward';
-    } else {
-      // For desktop devices, we use the position reported by the textarea control
-      selStart = ta1.selectionStart;
-      selLength = ta1.selectionEnd - ta1.selectionStart;
-      selDirection = ta1.selectionDirection;
-    }
+    // We use the position reported by the textarea control
```
Comment (mcdurdin) [inline developer/src/server/src/site/chargrid.js:76]: We'll need to test these changes on mobile devices as well as mobile-on-desktop mode, more in terms of making sure Developer test still works as expected without the touch alias stuff (I mean, it should, but focus changes could be relevant)

File: developer/src/server/src/site/test.js [hunk 3]
```diff
@@ -238,11 +238,10 @@ window.onload = function() {
   keyman.addEventListener('keyboardchange', function(keyboardProperties) {
     if(newOSK) {
       keyman.osk = newOSK;
-      newOSK.activeKeyboard = keyman.core.activeKeyboard;
+      newOSK.activeKeyboard = keyman.contextManager.activeKeyboard;
```
Comment (mcdurdin) [inline developer/src/server/src/site/test.js:241]:
```suggestion
      newOSK.activeKeyboard = keyman.contextManager.activeKeyboard;  // Private API refs on both sides
```

Are these still internal references? I believe `keyman.core.activeKeyboard` was not a published API?

**Review by mcdurdin — APPROVED** [review]

File: developer/src/server/build.sh (hunk 1 above)
Comment (jahorton) [reply developer/src/server/build.sh:127]: At the moment, KMW builds both debug and release configurations at the same time.  The testing pages all use the `--debug` variant too, so it should generally be built during standard `$KEYMAN_ROOT/web/build.sh` calls.  So... `--debug` is unnecessary for producing the non-minified version.  We could add it if you're worried about changing that in the future though, I guess.

I suppose I could have `--debug` disable the release build... but that's such a small fraction of the build time that it's insignificant.  I'm pretty sure each of the `esbuild` calls completes within a second, maybe two at most... and that's the only real point of divergence.

File: developer/src/server/src/site/chargrid.js (hunk 2 above)
Comment (jahorton) [reply developer/src/server/src/site/chargrid.js:76]: Would a simple, naive "try it from your phone on the same network" test based on one of the URLs put forth by Developer be enough, or is there more detail to consider when writing such a test?

File: developer/src/server/src/site/test.js (hunk 3 above)
Comment (jahorton) [reply developer/src/server/src/site/test.js:241]: I suppose they are.  That said, `newOSK.activeKeyboard` or something similar should eventually become "public" for a published, independent OSK module, though possibly after a proper API redesign phase.

File: developer/src/server/build.sh (hunk 1 above)
Comment (mcdurdin) [reply developer/src/server/build.sh:127]: It's kinda helpful as a mnemonic to remind readers of the script that Developer is using the unminified files, but not super important (and just keeping it consistent with other platform builds which don't build debug files as a step on the way to release builds).

File: developer/src/server/src/site/chargrid.js (hunk 2 above)
Comment (mcdurdin) [reply developer/src/server/src/site/chargrid.js:76]: Yes, can even test in Chrome's mobile device emulation.

File: developer/src/server/build.sh (hunk 1 above)
Comment (jahorton) [reply developer/src/server/build.sh:127]:
```suggestion
  ./build.sh build --debug
```

File: developer/src/server/src/site/chargrid.js (hunk 2 above)
Comment (jahorton) [reply developer/src/server/src/site/chargrid.js:76]: Documented as a TODO item on the main feature branch PR (#8560).

**Pull request merged**